agent/proxycfg/testing_terminating_gateway.go

```diff
@@ -596,19 +596,22 @@ func TestConfigSnapshotTerminatingGatewayHTTP2(t testing.T) *ConfigSnapshot {
 func TestConfigSnapshotTerminatingGatewaySubsetsHTTP2(t testing.T) *ConfigSnapshot {
 	web := structs.NewServiceName("web", nil)
 
-	return TestConfigSnapshotTerminatingGateway(t, false, nil, []UpdateEvent{
+	return TestConfigSnapshotTerminatingGateway(t, false, nil, []cache.UpdateEvent{
 		{
 			CorrelationID: serviceResolverIDPrefix + web.String(),
-			Result: &structs.ConfigEntryResponse{
-				Entry: &structs.ServiceResolverConfigEntry{
-					Kind: structs.ServiceResolver,
-					Name: "web",
-					Subsets: map[string]structs.ServiceResolverSubset{
-						"v1": {
-							Filter: "Service.Meta.version == 1",
-						},
-						"v2": {
-							Filter: "Service.Meta.version == 2",
+			Result: &structs.IndexedConfigEntries{
+				Kind: structs.ServiceResolver,
+				Entries: []structs.ConfigEntry{
+					&structs.ServiceResolverConfigEntry{
+						Kind: structs.ServiceResolver,
+						Name: "api",
```

Comment:
Does this matter for the expected result?
```suggestion
						Name: "web",
```